src/Processors/Formats/IRowInputFormat.cpp

```diff
@@ -98,36 +101,53 @@ Chunk IRowInputFormat::generate()
+                    ++num_errors;
+                    Float64 current_error_ratio = static_cast<Float64>(num_errors) / total_rows;
 
-                if (num_errors > params.allow_errors_num
-                    && current_error_ratio > params.allow_errors_ratio)
-                {
-                    e.addMessage("(Already have " + toString(num_errors) + " errors"
-                        " out of " + toString(total_rows) + " rows"
-                        ", which is " + toString(current_error_ratio) + " of all rows)");
-                    throw;
-                }
+                    if (num_errors > params.allow_errors_num
+                            && current_error_ratio > params.allow_errors_ratio)
+                    {
+                        e.addMessage("(Already have " + toString(num_errors) + " errors"
+                                " out of " + toString(total_rows) + " rows"
+                                ", which is " + toString(current_error_ratio) + " of all rows)");
+                        throw;
+                    }
 
-                if (!allowSyncAfterError())
-                {
-                    e.addMessage("(Input format doesn't allow to skip errors)");
-                    throw;
-                }
+                    if (!allowSyncAfterError())
+                    {
+                        e.addMessage("(Input format doesn't allow to skip errors)");
+                        throw;
+                    }
 
-                syncAfterError();
+                    syncAfterError();
 
-                /// Truncate all columns in block to initial size (remove values, that was appended to only part of columns).
+                    /// Truncate all columns in block to initial size (remove values, that was appended to only part of columns).
 
-                for (size_t column_idx = 0; column_idx < num_columns; ++column_idx)
+                    for (size_t column_idx = 0; column_idx < num_columns; ++column_idx)
+                    {
+                        auto & column = columns[column_idx];
+                        if (column->size() > num_rows)
+                            column->popBack(column->size() - num_rows);
+                    }
+                }
+                else
                 {
-                    auto & column = columns[column_idx];
-                    if (column->size() > num_rows)
-                        column->popBack(column->size() - num_rows);
+                    err_column->insert(e.message());
+                    auto current_num_rows = err_column->size();
+                    for (size_t column_idx = 0; column_idx < num_columns; ++column_idx)
+                    {
+                        auto & column = columns[column_idx];
+                        if (column->size() < current_num_rows)
+                        {
+                            column->insertDefault();
+                        }
+                    }
+                    syncAfterError();
```

Comment:
Can we skip `allowSyncAfterError` here? Is `allowSyncAfterError = false` and `append_error_column` incompatible ?